go/json2/unmarshal.go

```diff
@@ -33,7 +33,8 @@ var carriageReturn = []byte("\n")
 // efficient and should not be used for high QPS operations.
 func Unmarshal(data []byte, v any) error {
 	if pb, ok := v.(proto.Message); ok {
-		return annotate(data, protojson.Unmarshal(data, pb))
+		opts := protojson.UnmarshalOptions{DiscardUnknown: true}
+		return annotate(data, opts.Unmarshal(data, pb))
```

Comment:
This behavior more follows how protobuf itself works, which allows for fields to be removed and it ignores it in those cases.